labware-library/src/labware-creator/components/ConditionalLabwareRender.tsx

```diff
@@ -19,74 +20,107 @@ interface Props {
   definition: LabwareDefinition2 | null
 }
 
-const calculateViewBox = (args: {
-  bBox: DOMRect | undefined
-  xDim: number
-  yDim: number
-}): string => {
-  const { bBox, xDim, yDim } = args
-
+const calculateViewBox = (bBox: DOMRect): string => {
   // by-eye margin to make sure there is no visual clipping
   const MARGIN = 5
 
   // calculate viewBox such that SVG is zoomed and panned with the bBox fully in view,
   // in a "zoom to fit" manner, plus some visual margin to prevent clipping
-  return `${(bBox?.x ?? 0) - MARGIN} ${(bBox?.y ?? 0) - MARGIN} ${
-    xDim + MARGIN * 2
-  } ${yDim + MARGIN * 2}`
+  const x = bBox.x - MARGIN
+  const y = bBox.y - MARGIN
+  const xDimension = bBox.width + MARGIN * 2
+  const yDimension = bBox.height + MARGIN * 2
+  return `${x} ${y} ${xDimension} ${yDimension}`
+}
+
+const areBBoxesEqual = (
+  a: DOMRect | undefined,
+  b: DOMRect | undefined
```

Comment:
nit, what you have works. but we prefer to use `?` to signify that a prop can be undefined instead of typing out `undefined`. IDK if this is recommended in general but its the pattern we follow

```suggestion
  a?: DOMRec,
  b?: DOMRect,
```

Reply:
`| undefined` is different to TS than `?` in that it means someone must specify a `DOMRect | undefined` value explictly, whereas `?` means it's ok to omit the value.

Reply:
Interesting, thanks both for your feedback. I didn't know about the `?:` preference or the difference in how omitted values are handled.

This is moot now because it's `| null` now instead of `| undefined`.